src/index.coffee

```diff
@@ -0,0 +1,126 @@
+client = require './utils/client'
+
+# utility functions
+checkError = (error, response, body, code, callback) ->
+    callback errorMaker error, response, body, code
+
+errorMaker = (error, response, body, expectedCode) ->
+    if error
+        return error
+    else if response.status isnt expectedCode
+        msgStatus = "expected: #{expectedCode}, got: #{response.statusCode}"
+        err = new Error "#{msgStatus} -- #{body.error} -- #{body.reason}"
+        err.status = response.statusCode
+        return err
+    else
+        return null
+
+define = (docType, name, request, callback) ->
+    {map, reduce} = request
+
+    # transforms all functions in anonymous functions
```

Comment:
Maybe I'm wrong, but I think it transforms the functions in strings, not in anonymous functions.